docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,442 @@
+
+* `"eslint:recommended"`: the rules recommended by ESLint
+* `"eslint:all"`: all rules shipped with ESLint
+
+You can also use the `extends` property to extend a custom config. Custom configs can either be paths to local config files or npm package names.
+
+In flat config files, predefined configs are imported from separate modules into flat config files. The `recommended` and `all` rules configs are located in the [`@eslint/js`](https://www.npmjs.com/package/@eslint/js) package. You must import this package to use these configs:
+
+```shell
+npm install @eslint/js --save-dev
+```
+
+You can add each of these configs to the exported array or expose specific rules from them. You must import the modules for local config files and npm package configs with flat config.
+
+For example, here's an eslintrc file using the built-in `eslint:recommended` config:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: "eslint:recommended",
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+This eslintrc file uses built-in config, local custom config, and custom config from an npm package:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    extends: ["eslint:recommended", "./custom-config.js", "eslint-config-my-config"],
+    rules: {
+        semi: ["warn", "always"]
+    },
+    // ...other config
+}
+```
+
+To use the same configs in flat config, you would do the following:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customConfig from "./custom-config.js";
+import myConfig from "eslint-config-my-config";
+
+export default [
+    js.configs.recommended,
+    customConfig,
+    myConfig,
+    {
+        rules: {
+            semi: ["warn", "always"]
+        },
+        // ...other config
+    }
+];
+```
+
+Note that because you are just importing JavaScript modules, you can mutate the config objects before ESLint uses them. For example, you might want to have a certain config object only apply to your test files:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customTestConfig from "./custom-test-config.js";
+
+export default [
+    js.configs.recommended,
+    {
+        ...customTestConfig,
+        files: ["**/*.test.js"],
+    },
+];
+```
+
+### Ignoring Files
+
+With eslintrc, you can make ESLint ignore files by creating a separate `.eslintignore` file in the root of your project. The `.eslintignore` file uses the same glob pattern syntax as `.gitignore` files. Alternatively, you can use an `ignorePatterns` property in your eslintrc file.
+
+To ignore files with flat config, you can use the `ignores` property in a config object. The `ignores` property accepts an array of glob patterns. Note that flat config glob patterns do _not_ match dot files (e.g. `.dotfile.js`).
```

Comment:
Hi everyone,
Maybe we should clarify that [old .eslintignore no longer works with flat config](https://github.com/eslint/eslint/discussions/17342#discussioncomment-6359326)?
I'm not fully sure that it's understandable with the current ignore docs info
Just a small notion:

```suggestion
To ignore files with flat config, you can use the `ignores` property in a config object. The `ignores` property accepts an array of glob patterns. Note that flat config glob patterns do _not_ match dot files (e.g. `.dotfile.js`). Please note that since v8.25.0 the [new flat config system no longer supports .eslintignore files](https://eslint.org/blog/2022/10/eslint-v8.25.0-released/#highlights).
```

Reply:
Probably a good idea, though I'd prefer this:

```suggestion
To ignore files with flat config, you can use the `ignores` property in a config object. The `ignores` property accepts an array of glob patterns. Note that flat config glob patterns do _not_ match dot files (e.g. `*.js` won't match `.dotfile.js`). Flat config does not support loading ignore patterns from `.eslintignore` files, so you'll need to migrate those patterns directly into flat config.
```

The move away from `.eslintignore` was fairly early in the flat config lifecycle, so I don't think we need to mark it in time.

Reply:
> Probably a good idea, though I'd prefer this:
> 
> The move away from `.eslintignore` was fairly early in the flat config lifecycle, so I don't think we need to mark it in time.

Maybe you're right)
I started using flat config right after the release. And during updates, I had to figure out why .eslintignore no longer works for the flat config. And the only mention where I found it was in this blog post. So I think It's a little confusing. But I agree that this is a rare situation. And your correction fits

Reply:
adding your suggestion @nzakas